bentoml/handlers/json_handler.py

```diff
@@ -45,8 +50,42 @@ def handle_request(self, request, func):
         json_output = api_func_result_to_json(result)
         return Response(response=json_output, status=200, mimetype="application/json")
 
-    def handle_batch_request(self, requests, func):
-        raise NotImplementedError
+    def handle_batch_request(
+        self, requests: Iterable[SimpleRequest], func
+    ) -> Iterable[SimpleResponse]:
+        bad_resp = SimpleResponse(400, None, "Bad Input")
+        instances_list = [None] * len(requests)
+        responses = [bad_resp] * len(requests)
+
+        for i, request in enumerate(requests):
+            try:
+                raw_str = request.data
+                parsed_json = json.loads(raw_str)
+                instances_list[i] = parsed_json
+            except (json.JSONDecodeError, UnicodeDecodeError):
+                responses[i] = SimpleResponse(400, None, "not a valid json input")
+            except Exception:  # pylint: disable=broad-except
+                import traceback
+
+                traceback.print_exc()
+
+        merged_instances, slices = concat_list(instances_list)
+        merged_result = func(merged_instances)
+        if not isinstance(merged_result, (list, tuple)) or len(merged_result) != len(
+            merged_instances
+        ):
+            raise ValueError(
+                "The return value with JsonHandler must be list of jsonable objects, "
+                "and have same length as the inputs."
+            )
+
+        results = [merged_result[s] for s in slices]
+
+        for i, result in enumerate(results):
+            result_str = api_func_result_to_json(result)
+            responses[i] = SimpleResponse(200, dict(), result_str)
```

Comment:
isn't this going to rewrite the  `SimpleResponse(400, None, "not a valid json input")` above?

Reply:
maybe in line 58 initialize it as:
```
        responses = [None] * len(requests)
```
and here you can do something like:
```
if responses[I] is None:
   responses[i] = SimpleResponse(200, dict(), result_str)
```

Reply:
Oh, I see. I'll make sure these handlers are as robust as DataframeHandler did in later commits.